stdlib/privacy/src/main/ballerina/privacy/database_pii_store_utils.bal

```diff
@@ -77,18 +77,15 @@ function validateFieldName (string tableName, string idColumn, string piiColumn)
 # + queryResult - results of the insert query
 # + return - pseudonymized identifier if insert was successful, error if insert failed
 function processInsertResult (string id, int|error queryResult) returns string|error {
-    match queryResult {
-        int rowCount => {
-            if (rowCount > 0) {
-                return id;
-            } else {
-                error err = error("Unable to insert PII with identifier " + id);
-                return err;
-            }
-        }
-        error err => {
+    if (queryResult is int) {
+        if (queryResult > 0) {
+            return id;
+        } else {
+            error err = error("Unable to insert PII with identifier " + id);
             return err;
         }
+    } else {
```

Comment:
Can't we use check at the beginning and return the error there instead, with such functions?

```bal
         int intResult = check queryResult;
         if (intResult > 0) {
             return id;
         } else {
             error err = error("Unable to insert PII with identifier " + id);
             return err;
         }
```